dev-libs/lief/lief-0.11.5.ebuild

```diff
@@ -0,0 +1,125 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{8..10} pypy3 )
+
+# Upstream has two alternate approaches to building its Python API:
+# 1. A working "CMakeList.txt" only supporting a single Python target.
+# 2. A non-working "setup.py" supporting multiple Python targets but internally
+#    invoking CMake in mostly non-configurable (and thus broken) ways.
+# We choose the working approach.
+inherit python-single-r1 cmake
+
+DESCRIPTION="Library to instrument executable formats"
+HOMEPAGE="https://lief.quarkslab.com"
+SRC_URI="https://github.com/lief-project/LIEF/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="c examples +python static-libs"
```

Comment:
If GitHub's code-type detection is to be believed, the code is almost entirely c++ instead of plain c. In which case we should name this flag `cxx`, which also happens to be a global flag so in that case we can drop the flag description in `metadata.xml`: https://github.com/lief-project/LIEF

Reply:
> If GitHub's code-type detection is to be believed, the code is almost entirely c++ instead of plain c.

You remain correct in all things. This baffles me, but I've come to just accept it.

> In which case we should name this flag `cxx`...

I'd love to! I'd like nothing more than to simplify my life. But... doesn't the `cxx` USE flag govern C++ support? The `c` USE flag here actually governs LIEF's C support via `-DLIEF_C_API="$(usex c ON OFF)"`, as odd as a framework with dual C and C++ support seems to even me.

I couldn't actually find a comparable USE flag governing C support in `/usr/portage/profiles/use.desc`. So, I just went with `c`. Is that awful? I bicycled 30km today and am thus on the verge of physical and spiritual collapse, so I accept that I may currently have no idea what's going on here.

Reply:
> I'd love to! I'd like nothing more than to simplify my life. But... doesn't the cxx USE flag govern C++ support? The c USE flag here actually governs LIEF's C support via -DLIEF_C_API="$(usex c ON OFF)", as odd as a framework with dual C and C++ support seems to even me.

Right, I thought maybe the `-DLIEF_C_API` option toggled both the c++ part and the c part, but if it toggles only the c part, then I think `c` would indeed be the more appropriate name for the USE flag.